modin/pandas/groupby.py

```diff
@@ -817,11 +829,13 @@ def _index_grouped(self):
         if self._index_grouped_cache is None:
             if hasattr(self._by, "columns") and len(self._by.columns) > 1:
                 by = list(self._by.columns)
-                is_multi_by = True
             else:
                 by = self._by
-                is_multi_by = self._is_multi_by
-            if is_multi_by:
+            level = [
+                self._query_compiler.get_axis(self._axis).get_level_values(o)
+                for o in self._level
+            ]
```

Comment:
```python
df.groupby("level_name").groups != df.index.groupby("level_name") # previous logic
df.groupby("level_name").groups == df.index.groupby(df.index.levels["level_name"]) # current logic
```
we're using it here:
https://github.com/modin-project/modin/blob/bf741e6f8c266f0cec2525d21c1a4fe219a7fc38/modin/pandas/groupby.py#L851